### ISSUE TITLE
PrepareInventoryResultEvent deprecated 

### ISSUE THREAD

**Issue by SlimeDog** [PimvanderLoos/ArmoredElytra#11]
Paper 1.19.3-328
ArmoredElytra 2.4.15

PrepareInventoryResultEvent deprecated 

On server start-up
```
[Server thread/WARN]: [ArmoredElytra] "ArmoredElytra v2.4.15" has registered a listener for org.bukkit.event.inventory.PrepareInventoryResultEvent on method "private void nl.pim16aap2.armoredElytra.handlers.EventHandlers.onAnvilInventoryOpen(org.bukkit.event.inventory.PrepareAnvilEvent)", but the event is Deprecated. "Server performance will be affected"; please notify the authors [pim16aap2].
```

**Comment by PimvanderLoos** [comment]
The PrepareInventoryResultEvent is only [deprecated for paper](https://jd.papermc.io/paper/1.19/org/bukkit/event/inventory/PrepareInventoryResultEvent.html), while [it isn't for Spigot](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/event/inventory/PrepareInventoryResultEvent.html).

As can be read in the deprecation message, Paper suggests replacing `PrepareInventoryResultEvent` with [PrepareResultEvent](https://jd.papermc.io/paper/1.19/com/destroystokyo/paper/event/inventory/PrepareResultEvent.html). However, `PrepareResultEvent` only exists in Paper; not in Spigot.
As such, I cannot simply replace the 'old' event with the 'new' one without breaking support for all Spigot servers. 

Of course, I could write code to load an alternative class in case the 'new' event exists. However, looking at [the code](https://github.com/PaperMC/Paper/blob/master/patches/api/0213-Add-PrepareResultEvent-PrepareGrindstoneEvent.patch) of the 'new' event, it doesn't look like it offers anything of value at all, much less something to justify the added work/complexity to support basically the exact same event with a slightly different name.

I'm also not entirely sure why they say that server performance will be affected. As far as I can see, the only difference between the two events is that in Paper's version, `PrepareResultEvent` and its subclasses dump all their listeners in a single handlers list, whereas the Spigot version has a handlers list for each type of `PrepareInventoryResultEvent`. I don't really see how that could degrade server performance. There are, after all, already a great many handler lists used on the server; most event types have one. On top of that, I'm not sure if pooling multiple event types in a single handler list will result in a decrease or and increase in performance. I suspect that the exact impact depends on the situation, but will be negligible regardless.
Furthermore, `PrepareResultEvent` is a subclass of `PrepareInventoryResultEvent`, and the only methods it overrides call their supermethod anyway, meaning it doesn't really do anything that the 'old' event did as well.
Lastly, the [PrepareAnvilEvent](https://jd.papermc.io/paper/1.19/org/bukkit/event/inventory/PrepareAnvilEvent.html) (which is the specific event I'm using) [is already a subclass](https://github.com/PaperMC/Paper/blob/c919e944ffedf06a6a5625043bc744b06e499442/patches/api/0213-Add-PrepareResultEvent-PrepareGrindstoneEvent.patch#L95-L96) of the `PrepareResultEvent`, so I'm not sure what the point of all this is.

So, long story short, unless I am missing something, I do not see any reason to modify the plugin. Instead, I think it would make more sense to update paper and remove the, as far as I can see, overzealous error message.

**Comment by SlimeDog** [comment]
Confirmed fixed by Paper 1.19.3-330. Thanks to **blablubbabc** for encouraging them to fix it.
